price-discovery/src/lib.rs

```diff
@@ -50,7 +45,6 @@ pub trait PriceDiscovery:
             "Start epoch must be before end epoch"
         );
 
-        self.dex_sc_address().set(&dex_sc_address);
         self.launched_token_id().set(&launched_token_id);
```

Comment:
do we already have an only owner function which can set the dex sc address ?

Reply:
I added an only_owner function for this set with this PR